odoo/upgrade_code/18.1-00-sql-constraint.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+import ast
+import json
+import logging
+import re
+import typing
+
+if typing.TYPE_CHECKING:
+    from odoo.cli.upgrade_code import FileManager
+
+
+def upgrade(file_manager: FileManager):
+    log = logging.getLogger(__name__)
+    sql_expression_re = re.compile(r"_sql_constraints\s*=\s*\[([^\]]+)]")
```

Comment:
Suggestion: make it a global constant, like `SQL_CONSTRAINTS_RE`.
Also, you should match it as a single identifier (either use `\b` or match the beginning of the line.)
You may use a named group to improve code readability, like `r"...=\s*(P<list>\[[^\]]+\])"`.

Reply:
I have added `\b`.
Creating a constant here does not add any value, since the script is executed only once anyways.
I find named groups to be less readable when there is 1 (or sometimes a few without alternatives) capturing groups.